docs/development/design/embed-api.rst

```diff
@@ -0,0 +1,154 @@
+      }
+
+Get section
+-----------
+
+Allow us to query the content of the section, with all links re-written as absolute.
+
+.. http:get:: /_/api/v3/embed/sections?project=docs&version=latest&path=install.html#examples
+
+   :query project: (required)
+   :query version: (required)
+   :query path: Path with or without fragment (required)
+
+   .. sourcecode:: json
+
+      {
+         "project": "docs",
+         "version": "latest",
+         "path": "install.html",
+         "url": "https://docs.readthedocs.io/en/latest/install.html#examples",
+         "id": "examples",
+         "title": "Examples",
+         "content": "<div>I'm a html block!<div>",
+         "extras": {
+            "js": ["https://docs.readthedocs.io/en/latest/index.js"],
+            "css": ["https://docs.readthedocs.io/en/latest/index.css"],
+         }
+      }
+
+Notes
+-----
+
+- If a section or page doesn't exist, we return 404.
+- All links are re-written to be absolute (this is already done).
+- All sections listed are from html tags that are linkeable, this is, they have an ``id``
+  (we don't rely on the toctree from the fjson file anymore).
+- The IDs returned don't contain the redundant ``#`` symbol.
+- The content is an string with a well formed HTML block.
+- We could also support only ``url`` as argument for ``/sections`` and ``/pages``,
+  but this introduces another way of querying the API.
+  Having two ways of querying the API makes it *less-cacheable*.
+- Returning the extra js and css requires parsing the HTML page itself,
+  rather than only the content extracted from the fjson files (this is for sphinx).
+  We can use both, the html file and the json file, but we could also just start parsing the full html page
+  (we can re-use code from the search parsing to detect the main content).
+- ``extras`` could be returned only on ``/pages``, or only on ``/sections``.
+  It makes more sense to be only on ``/pages``,
+  but them querying a section would require to query a page to get the extra js/css files.
+- We could not return the ``title`` of the page/section as it would require more parsing to do
+  (but we can re-use the code from search).
+  Titles can be useful to build an UI like https://readthedocs.org/projects/docs/tools/embed/.
+- MkDocs support can be added easily as we make our parsing code more general.
+
+.. note::
+
+   We should probably make a distinction between our general API that handles Read the Docs resources,
+   vs our APIs that expose features (like server side search, footer, and embed, all of them proxied).
+   This way we can version each endpoint separately.
+
+Deprecation
+-----------
+
+We should have a section in our docs instead of guide where the embed API is documented.
+There we can list v2 as deprecated.
+We would need to migrate our extension as well.
+Most of the parsing code could be shared between the two APIs, so it shouldn't be a burden to maintain.
+
+API Client
+----------
+
+Do we really need a JS client?
```

Comment:
I think so. IIRC @agjohnson idea here was to include the popup/modal form of this into the JS client. Besides, we could eventually be able to use this JS client from our extension itself, instead of maintaining the front-end in our own extension (or adding it as another popup extension together with the ones we currently support)

Reply:
Yea, the JS client is going to be doing a lot of work in addition to just doing the GET. We definitely want nice default widgets, along with helper methods that make this a much nicer UX for users.